File: c#/src/Docmost.Sdk/Services/SpacesApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class SpacesApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetWorkspaceSpacesAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetSpaceInfoAsync(SpaceIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CreateSpaceAsync(CreateSpaceDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateSpaceAsync(UpdateSpaceDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeleteSpaceAsync(SpaceIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/delete", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetSpaceMembersAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/members", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> AddSpaceMembersAsync(AddSpaceMembersDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/members/add", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> RemoveSpaceMemberAsync(RemoveSpaceMemberDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/members/remove", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateSpaceMemberRoleAsync(UpdateSpaceMemberRoleDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("spaces/members/change-role", request, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/spaces-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { AddSpaceMembersDto, CreateSpaceDto, PaginationOptions, RemoveSpaceMemberDto, SpaceIdDto, UpdateSpaceDto, UpdateSpaceMemberRoleDto } from '../models/index.js';


export class SpacesApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getWorkspaceSpaces(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces`, request);
  }

  async getSpaceInfo(request: SpaceIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/info`, request);
  }

  async createSpace(request: CreateSpaceDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/create`, request);
  }

  async updateSpace(request: UpdateSpaceDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/update`, request);
  }

  async deleteSpace(request: SpaceIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/delete`, request);
  }

  async getSpaceMembers(): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/members`, undefined);
  }

  async addSpaceMembers(request: AddSpaceMembersDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/members/add`, request);
  }

  async removeSpaceMember(request: RemoveSpaceMemberDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/members/remove`, request);
  }

  async updateSpaceMemberRole(request: UpdateSpaceMemberRoleDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`spaces/members/change-role`, request);
  }

}
